include/zephyr/net/mqtt.h

```diff
@@ -209,8 +235,86 @@ struct mqtt_connack_param {
 
 	/** The appropriate non-zero Connect return code indicates if the Server
 	 *  is unable to process a connection request for some reason.
+	 *  MQTT 3.1 - Return codes specified in @ref mqtt_conn_return_code
+	 *  MQTT 5.0 - Reason codes specified in @ref mqtt_connect_reason_code
 	 */
-	enum mqtt_conn_return_code return_code;
+	uint8_t return_code;
+
+#if defined(CONFIG_MQTT_VERSION_5_0)
+	struct {
+		/** MQTT 5.0, ch. 3.2.2.3.10 User Property. */
+		struct mqtt_utf8_pair user_prop[CONFIG_MQTT_USER_PROPERTIES_MAX];
+
+		/** MQTT 5.0, ch. 3.2.2.3.7 Assigned Client Identifier. */
+		struct mqtt_utf8 assigned_client_id;
+
+		/** MQTT 5.0, ch. 3.2.2.3.9 Reason String. */
+		struct mqtt_utf8 reason_string;
+
+		/** MQTT 5.0, ch. 3.2.2.3.15 Response Information. */
+		struct mqtt_utf8 response_information;
+
+		/** MQTT 5.0, ch. 3.2.2.3.16 Server Reference. */
+		struct mqtt_utf8 server_reference;
+
+		/** MQTT 5.0, ch. 3.2.2.3.17 Authentication Method. */
+		struct mqtt_utf8 auth_method;
+
+		/** MQTT 5.0, ch. 3.2.2.3.18 Authentication Data. */
+		struct mqtt_binstr auth_data;
+
+		/** MQTT 5.0, ch. 3.2.2.3.2 Session Expiry Interval. */
+		uint32_t session_expiry_interval;
+
+		/** MQTT 5.0, ch. 3.2.2.3.6 Maximum Packet Size. */
+		uint32_t maximum_packet_size;
+
+		/** MQTT 5.0, ch. 3.3.2.3.3 Receive Maximum. */
+		uint16_t receive_maximum;
+
+		/** MQTT 5.0, ch. 3.2.2.3.8 Topic Alias Maximum. */
+		uint16_t topic_alias_maximum;
+
+		/** MQTT 5.0, ch. 3.2.2.3.14 Server Keep Alive. */
+		uint16_t server_keep_alive;
+
+		/** MQTT 5.0, ch. 3.2.2.3.4 Maximum QoS. */
+		uint8_t maximum_qos;
+
+		/** MQTT 5.0, ch. 3.2.2.3.5 Retain Available. */
+		uint8_t retain_available;
+
+		/** MQTT 5.0, ch. 3.2.2.3.11 Wildcard Subscription Available. */
+		uint8_t wildcard_sub_available;
+
+		/** MQTT 5.0, ch. 3.2.2.3.12 Subscription Identifiers Available. */
+		uint8_t subscription_ids_available;
+
+		/** MQTT 5.0, ch. 3.2.2.3.13 Shared Subscription Available. */
+		uint8_t shared_sub_available;
+
+		/** Flags indicating whether given property was present in received packet. */
+		struct {
+			bool has_session_expiry_interval;
```

Comment:
There are quite many bool fields, should we make them as bitfields to save some space?
The is a doxygen comment for this struct, and the fields are self documenting, but the output looks strange
https://builds.zephyrproject.io/zephyr/pr/87354/docs/doxygen/html/structmqtt__connack__param.html#details
not sure what to do here.

Reply:
Btfields were indeed my first choice, however when I worked on the decoder it turned out to be a problem, as bitfields are not addressable, hence I wasn't able to create a single generic decoder function for properties, the code was a bit bloated with bitfields. I'm open for ideas though on how could that be tackled differently.

I'll check what's going on with the docs.

Reply:
It seems doxygen interprets `ch.` as end of sentence and breaks the line, I'll fix that. And possibly add separate docstrings for `has_*` fields so they don't look empty.